src/platform/Linux/DnssdImpl.cpp

```diff
@@ -662,6 +685,26 @@ void CopyTypeWithoutProtocol(char (&dest)[N], const char * typeAndProtocol)
     }
 }
 
+void MdnsAvahi::BrowseRetryCallback(chip::System::Layer * aLayer, void * appState)
+{
+    BrowseContext * context = static_cast<BrowseContext *>(appState);
+    // Don't schedule anything new if we've stopped.
+    if (context->mStopped.load())
+    {
+        return;
```

Comment:
How is `context` deleted if a browser is not created (since it seems to me like AVAHI_BROWSER_ALL_FOR_NOW handler is the one that deletes the context)

maybe we should have:

```cpp
AvahiServiceBrowser * newBrowser = nullptr;
if (!context->mStopped.load())
{
   newBrowser = avahi_service_browser_new(  /* */ );
}

if (newBrowser == nullptr)
{
   // ...
}
```

Reply:
Added a fix and some comments. Cancelling the timer was also an issue.